WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -59,6 +65,41 @@ class MainActivity : AppCompatActivity() {
 
     }
 
+    /** Instantiate the Listener  */
+    class MyListener() : WebViewCompat.WebMessageListener {
+        override fun onPostMessage(
+            view: WebView,
+            message: WebMessageCompat,
+            sourceOrigin: Uri,
+            isMainFrame: Boolean,
+            replyProxy: JavaScriptReplyProxy
+        ) {
+            replyProxy.postMessage("Got it!")
+            Log.i("grcoleman", "received onPostMessage in app")
+        }
+    }
+
+    /** Instantiate a class to determine which JS-Java API to use based on the version the application is running on */
+    class JsObject() {
+        fun createJsObject(webview : WebView , context: Context) {
+            // If the application is running on a version that supports WebMessageListener then use it.
+            // WebMessageListener is preferred because it is more secure
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
```

Comment:
Based off of this [page](https://developer.android.com/jetpack/androidx/releases/webkit), should it be Lollipop?

Reply:
The minSdkVersion for the sample is Lollipop (API 21) so there's no need to check :)

Reply:
You shouldn't need an SDK version check regardless because androidx.webkit APIs are not guarded by SDK. Instead, our aPIs are guarded by `WebViewFeature.isFeatureSupported()` ([this doc](https://developer.android.com/reference/androidx/webkit/package-summary) has examples), which you already have below.

androidx.webkit's [minSdkVersion is 14](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-dev:buildSrc/src/main/kotlin/androidx/build/SupportConfig.kt;l=24;drc=6f347d71a6ef1fc135293f6f004b8ed9974ca752), but that number just determines whether your app can compile or not (our app's minSdk is larger, so there's no compilation issue), it's not a value you need to check at runtime.